kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb

```diff
@@ -0,0 +1,156 @@
+require 'httpclient'
+
+module KPM
+  module NexusFacade
+
+    class UnexpectedStatusCodeException < StandardError
+      def initialize(code)
+        @code = code
+      end
+
+      def  message
+        "The server responded with a #{@code} status code which is unexpected."
+      end
+    end
+
+    class ArtifactMalformedException < StandardError
+      class << self
+        def message
+          'Please submit your request using 4 colon-separated values. `groupId:artifactId:version:extension`'
+        end
+      end
+    end
+
+    # This is an extract and slim down of functions needed from nexus_cli to maintain the response expected by the base_artifact.
+    class NexusApiCallsV2
+      PULL_ARTIFACT_ENDPOINT = 'service/local/artifact/maven/redirect'
+      GET_ARTIFACT_INFO_ENDPOINT = 'service/local/artifact/maven/resolve'
+      SEARCH_FOR_ARTIFACT_ENDPOINT = 'service/local/data_index'
+
+      attr_reader :version
+      attr_reader :configuration
+      attr_reader :ssl_verify
+
+      def initialize(configuration, ssl_verify)
+        @configuration = configuration
+        @ssl_verify = ssl_verify
+      end
+
+      def search_for_artifacts(coordinates)
+        http_client = get_client
+        url = get_url(SEARCH_FOR_ARTIFACT_ENDPOINT, configuration)
+        group_id, artifact_id = coordinates.split(":")
+        response = http_client.get(url, :query => {:g => group_id, :a => artifact_id})
+
+        case response.status
+          when 200
+            return response.content
+          else
+            raise UnexpectedStatusCodeException.new(response.status)
+        end
+      end
+
+      def get_artifact_info(coordinates)
+        http_client = get_client
+        query_params = get_query_params(coordinates)
+        url = get_url(GET_ARTIFACT_INFO_ENDPOINT, configuration)
+        response = http_client.get(url,query_params)
+
+        case response.status
+          when 200
+            return response.content
+          when 404
+            raise StandardError.new('The artifact you requested information for could not be found. Please ensure it exists inside the Nexus.')
+          when 503
+            raise StandardError.new('Could not connect to Nexus. Please ensure the url you are using is reachable.')
+          else
+            raise UnexpectedStatusCodeException.new(response.status)
+        end
+      end
+
+      def pull_artifact(coordinates ,destination)
+        file_name = get_file_name(coordinates)
+        destination = File.join(File.expand_path(destination || "."), file_name)
+        http_client = get_client
+        query_params = get_query_params(coordinates)
+        url = get_url(PULL_ARTIFACT_ENDPOINT, configuration)
+        response = http_client.get(url,query_params)
```

Comment:
It would be nice to have a logger available at this point, and log the request and response in `debug`, in case we need to debug this.